control-plane/csi-driver/src/bin/controller/client.rs

```diff
@@ -498,4 +498,35 @@ impl RestApiClient {
             .await?;
         Ok(response.into_body())
     }
+
+    /// Expand volume.
+    #[instrument(fields(volume.uuid = %volume_id, volume.size = %required_volume_size), skip(self, volume_id, required_volume_size))]
+    pub(crate) async fn expand_volume(
+        &self,
+        volume_id: &uuid::Uuid,
+        required_volume_size: u64,
+    ) -> Result<Volume, ApiClientError> {
+        let vol_client = self.rest_client.volumes_api();
+
+        let vol = vol_client
+            .get_volume(volume_id)
+            .await
+            .map(|response| response.into_body())?;
+
+        // This call should be idempotent and should return success if the volume
+        // size is already greater than or equal to the target_volume_size.
+        if vol.spec.size.ge(&required_volume_size) {
```

Comment:
IIRC this can be done using the error code from put itself?

Reply:
Yes, I've refactored the idempotency implementation. WDYT?